apps/website/docs/react/api/merge-classes.md

```diff
@@ -0,0 +1,89 @@
+---
+sidebar_position: 2
+---
+
+import OutputTitle from '@site/src/components/OutputTitle'
+
+# mergeClasses
+
+There are cases where you need to merge classes from multiple `useClasses` calls. To properly merge the classes, you need to use `mergeClasses()` function, which performs merge and deduplication of atomic classes generated by `makeStyles()`.
+
+```jsx
+import { mergeClasses, makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  blueBold: {
+    color: 'blue',
+    fontWeight: 'bold',
+  },
+  red: {
+    color: 'red',
+  },
+});
+
+function Component(props) {
+  const { isBold } = props;
+  const classes = useClasses();
+
+  // ℹ️ Order of arguments determines the results
+
+  const firstClassName = mergeClasses(isBold && classes.blueBold, classes.red);
+  // 👆 { color: 'red', fontWeight: 'bold' }
+  const secondClassName = mergeClasses(classes.red, isBold && classes.blueBold);
+  // 👆 { color: 'blue', fontWeight: 'bold' }
+
+  return (
+    <>
+      <div className={firstClassName} />
+      <div className={secondClassName} />
+    </>
+  );
+}
+```
+
+:::note
+
+Compared to native CSS order of arguments in `mergeClasses()` determines the results. This allows to control styles with JavaScript.
+
+:::
+
+## Incorrect usages
+
+This section shows and describes anti-patterns which should never be used.
+
+### Never concatenate class strings
+
+:::caution
+
+It is not possible to simply concatenate classes returned by `useClasses()` hooks. Always use `mergeClasses()` to perform classes merging.
+
+:::
+
+```jsx
+const useClasses = makeStyles({
+  rootA: { color: 'red' },
+  rootB: { color: 'blue' },
+});
+
+function App(props) {
+  const classes = useClasses();
+  // ⚠️ Never concatenate class strings, always use mergeClasses()
+  const wrongClasses = classes.rootA + ' ' + classes.rootB;
+}
+```
+
+### Order of arguments matters
+
+There might be an issue with incorrect order of classes - as the latest wins. In the example below `props.className` should be last to override the component styles:
```

Comment:
```suggestion
The order of classes is important - If two classes apply the same style, the latest class wins. In the example below the background color will be red:
```

Let's not use a library component specific example, simple red/blue will be much clearer